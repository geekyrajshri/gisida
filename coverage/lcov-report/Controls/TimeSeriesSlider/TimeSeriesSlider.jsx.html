<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Controls/TimeSeriesSlider/TimeSeriesSlider.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">Controls/TimeSeriesSlider</a> TimeSeriesSlider.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
require('./TimeSeriesSlider.scss');
&nbsp;
export default function TimeSeriesSlider(props) {<span class="branch-0 cbranch-no" title="branch not covered" ></span>
  return (
    &lt;div className="series"&gt;
      &lt;label id={`${props.mapId}-label`} className="label" htmlFor="slider" /&gt;
      &lt;input id={`${props.mapId}-slider`} className="slider" type="range" list="datalist" /&gt;
      &lt;datalist id="datalist"&gt;
        {props.periods.map((p, index) =&gt;
          &lt;option key={index}&gt;{index}&lt;/option&gt;)}
      &lt;/datalist&gt;
    &lt;/div&gt;
  );
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 27 2017 15:27:56 GMT+0300 (EAT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
