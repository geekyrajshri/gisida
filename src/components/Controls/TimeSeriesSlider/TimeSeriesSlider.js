require('./TimeSeriesSlider.scss');

export default function TimeSeriesSlider(props) {
  return (
    <div className="series">
      <label id={props.mapId + '-label'} className="label" htmlFor="slider" />
      <input id={props.mapId + '-slider'} className="slider" type="range" list="datalist" />
      <datalist id="datalist">
        {props.periods.map((p, index) =>
          <option key={index}>{index}</option>)}
      </datalist>
    </div>
  );
}
