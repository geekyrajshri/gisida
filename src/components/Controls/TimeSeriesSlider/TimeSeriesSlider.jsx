import React from 'react';

require('./TimeSeriesSlider.scss');

function TimeSeriesSlider(props) {
  return (
    <div className="series">
      <label id={`${props.mapId}-label`} className="label" htmlFor="slider" />
      <input id={`${props.mapId}-slider`} className="slider" type="range" list="datalist" />
      <datalist id="datalist">
        {props.periods.map((p, i) =>
          <option key={i}>{i}</option>)}
      </datalist>
    </div>
  );
}

TimeSeriesSlider.propTypes = {
  mapId: React.PropTypes.string.isRequired,
  periods: React.PropTypes.arrayOf(React.PropTypes.any).isRequired,
};

export default TimeSeriesSlider;
