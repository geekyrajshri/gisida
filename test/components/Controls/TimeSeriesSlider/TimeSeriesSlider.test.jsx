import React from 'react';
import renderer from 'react-test-renderer';
import TimeSeriesSlider from '../../../../src/components/Controls/TimeSeriesSlider/TimeSeriesSlider';

describe('TimeSeriesSlider', () => {
    const periods = ['Week 1', 'Week 2', 'Week 3'];
    it('TimeSeriesSlider renders time series slider', () => {
        const component = renderer.create(<TimeSeriesSlider mapId={1} periods={periods} />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});
