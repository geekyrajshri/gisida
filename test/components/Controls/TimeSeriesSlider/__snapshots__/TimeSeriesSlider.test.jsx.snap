// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSeriesSlider TimeSeriesSlider renders time series slider 1`] = `
<div
  className="series"
>
  <label
    className="label"
    htmlFor="slider"
    id="1-label"
  />
  <input
    className="slider"
    id="1-slider"
    list="datalist"
    type="range"
  />
  <datalist
    id="datalist"
  >
    <option>
      0
    </option>
    <option>
      1
    </option>
    <option>
      2
    </option>
  </datalist>
</div>
`;
